core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -753,20 +753,17 @@ object TestUtils extends Logging {
     new ProducerRequest(correlationId, clientId, acks.toShort, timeout, collection.mutable.Map(data:_*))
   }
 
-  def makeLeaderForPartition(zkUtils: ZkUtils,
+  def makeLeaderForPartition(zkClient: KafkaZkClient,
                              topic: String,
                              leaderPerPartitionMap: scala.collection.immutable.Map[Int, Int],
                              controllerEpoch: Int) {
     leaderPerPartitionMap.foreach { case (partition, leader) =>
       try {
-        val newLeaderAndIsr = zkUtils.getLeaderAndIsrForPartition(topic, partition)
-          .map(_.newLeader(leader))
-          .getOrElse(LeaderAndIsr(leader, List(leader)))
-
-        zkUtils.updatePersistentPath(
-          getTopicPartitionLeaderAndIsrPath(topic, partition),
-          zkUtils.leaderAndIsrZkData(newLeaderAndIsr, controllerEpoch)
-        )
+        val leaderIsrEpoch = zkClient.getTopicPartitionState(new TopicPartition(topic, partition))
+        val leaderAndIsr = if (leaderIsrEpoch.isDefined) leaderIsrEpoch.get.leaderAndIsr else LeaderAndIsr(leader, List(leader))
```

Comment:
Use `Option.getOrElse`. In general, `isDefined` followed by `get` should never be done in Scala. There are other cases like this in this PR.

Reply:
@ijuma  updated the PR. Pls take a look